Dispatcher/UI/Homepage/HomepageViewController.swift

```diff
@@ -8,26 +8,40 @@
 import UIKit
 import Alamofire
 
-class HomeViewController: UIViewController {
-    
-    @IBOutlet weak var searchButton: UIButton!
-    @IBOutlet weak var notificationsButton: UIButton!
+class HomepageViewController: UIViewController {
+
     @IBOutlet weak var articlesTableView: UITableView!
     @IBOutlet var homepageView: UIView!
     
-    let homepageViewModel = HomepageViewModel()
+    let homepageViewModel = HomepageViewModel.shared
     let activityIndicator = UIActivityIndicatorView()
+    var headerView: HeaderView?
     
     override func viewDidLoad() {
         super.viewDidLoad()
         self.navigationController?.isNavigationBarHidden = true
         
+        headerView = HeaderView(frame: CGRect(x: 0, y: 0, width: homepageView.frame.width, height: 95), headerType: .mainHeader)
+        homepageView.addSubview(headerView ?? UIView())
+        setHeaderViewConstraints()
         setupTableView()
         configureActivityIndicator()
         getArticles()
     }
     
+    private func setHeaderViewConstraints() {
+        headerView!.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            headerView!.leadingAnchor.constraint(equalTo: homepageView.leadingAnchor, constant: 0),
```

Comment:
This relates to the comment above -  if you add the view inside the interface builder then you can already add all the constrains over there and have less code inside your VC. Also, personally I think it's very convenient to be able to see the constraints marked on the storyboard as it helps visualize them better in case they need to be changed